test/python/transpiler/test_decompose.py

```diff
@@ -21,6 +21,39 @@
 from qiskit.quantum_info.operators import Operator
 from qiskit.test import QiskitTestCase
 
+# example complex circuit
+#       ┌────────┐               ┌───┐┌─────────────┐
+# q2_0: ┤0       ├────────────■──┤ H ├┤0            ├
+#       │        │            │  └───┘│  circuit-57 │
+# q2_1: ┤1 gate1 ├────────────■───────┤1            ├
+#       │        │┌────────┐  │       └─────────────┘
+# q2_2: ┤2       ├┤0       ├──■──────────────────────
+#       └────────┘│        │  │
+# q2_3: ──────────┤1 gate2 ├──■──────────────────────
+#                 │        │┌─┴─┐
+# q2_4: ──────────┤2       ├┤ X ├────────────────────
+#                 └────────┘└───┘
+circ1 = QuantumCircuit(3)
+circ1.h(0)
+circ1.t(1)
+circ1.x(2)
+my_gate = circ1.to_gate(label="gate1")
+circ2 = QuantumCircuit(3)
+circ2.h(0)
+circ2.cx(0, 1)
+circ2.x(2)
+my_gate2 = circ2.to_gate(label="gate2")
+circ3 = QuantumCircuit(2)
+circ3.x(0)
+q_bits = QuantumRegister(5)
+qc = QuantumCircuit(q_bits)
+qc.append(my_gate, q_bits[:3])
+qc.append(my_gate2, q_bits[2:])
+qc.mct(q_bits[:4], q_bits[4])
+qc.h(0)
+qc.append(circ3, [0, 1])
+COMPLEX_CIRC = qc
```

Comment:
Could you move this inside the `TestDecompose` class? For objects we want to re-use during the test you can define them in the `setUp` method, e.g.
```python
class TestDecompose(QiskitTestCase):
    def setUp(self):
        super().setUp()
        # define circuit here...
        self.complex_circuit = qc
        
    def some_test(self):
        # use your circuit as self.complex_circuit

```